src/meta/src/stream/stream_graph.rs

```diff
@@ -603,9 +603,22 @@ impl StreamGraphBuilder {
                         }
                     }
 
-                    NodeBody::TopN(node) | NodeBody::AppendOnlyTopN(node) => {
-                        node.table_id += table_id_offset;
-                        ctx.internal_table_id_set.insert(node.table_id);
+                    NodeBody::TopN(node) => {
+                        node.table_id_l += table_id_offset;
+                        node.table_id_m += table_id_offset;
+                        node.table_id_h += table_id_offset;
+                        ctx.internal_table_id_set.insert(node.table_id_l);
+                        ctx.internal_table_id_set.insert(node.table_id_m);
+                        ctx.internal_table_id_set.insert(node.table_id_h);
+                    }
+
+                    NodeBody::AppendOnlyTopN(node) => {
```

Comment:
For append_only top_n, it use two internal state `table_id_l` and `table_id_h`, I'm not sure inserting two or three `table_id` for `append_only` `top_n`.
cc @st1page

Reply:
I think there should be two table id. but it is ok now that we waste one(not affect correctness). and later when we infer the table catalog in fe we can fix it.